lib/SILOptimizer/Mandatory/TFDeabstraction.cpp

```diff
@@ -2522,13 +2524,41 @@ void TFDeabstractionPass::run() {
                                    fn.getName().str().c_str());
 
     TFDeabstraction(fn, tfc, constantEvaluator, PM).doIt();
+    partitionedFunctions.insert(&fn);
 
     // TODO(clattner): This should eventually be the driver that kicks off
     // the partitioning pass as part of it, and the partitioning and later
     // passes are just function passes that are invoked by this one.  Until
     // we are ready for that, let them run later in the pipeline after the
     // other optimization and cleanup passes.
   }
+
+  // Deabstract stragglers that were left out in the previous iteration. These
+  // are functions that are *still* referred to in the code and operate on
+  // tensor values, but have not been partitioned. It can happen in the
+  // following case, for instance, where `foo` is an external function that
+  // takes a for which we do not have the body:
+  //   main() {
+  //     foo() { $0 -= 0.1 * $1 }
```

Comment:
can we turn something like this into a unit test?

Reply:
Added a unit test in deabstraction_finished.swift